### DISCUSSION TITLE
Multiline comments collapse to one line

### DISCUSSION THREAD

**Discussion by wa8f7192** [eclipse-langium/langium#1695] [Q&A] [answered]
Hi,

I would like to know if it is possible to collapse multiline comments to one line instead of two lines?
is this something we would need to adjust from the code editor instead / we can update the grammar accordingly?

![image](https://github.com/user-attachments/assets/ca34d059-408d-49bd-b311-07aa80e6cfb9)

![image](https://github.com/user-attachments/assets/2b6adb52-fb72-406e-8195-6e59077e56e4)

this is the code currently we have for comments

hidden terminal ML_COMMENT: /(\/\*[\s\S]*?\*\/)/;
hidden terminal ML_COMMENT1: '(*' -> '*)';
hidden terminal SL_COMMENT: /\/\/[^\n\r]*/;

**Comment by cdietrich** [comment]
Hi, did you check if adjusting
toFoldingRange in DefaultFoldingRangeProvider does work?

**Reply by msujew** [reply, marked as answer]
See also the `includeLastFoldingLine` method that explicitly controls this behavior:

https://github.com/eclipse-langium/langium/blob/335f9a99b1568829cd1c0f5d3bbe33ca754daf90/packages/langium/src/lsp/folding-range-provider.ts#L138-L149

@wa8f7192 You can override the class to change the behavior according to your liking.

**Reply by wa8f7192** [reply]
Hi cdietrich, msujew, thanks for the reply,  I will take a look at this.
